test/compilable/test16492.d

```diff
@@ -0,0 +1,12 @@
+// https://issues.dlang.org/show_bug.cgi?id=16492
+
+void mayCallGC();
+
+void test() @nogc pure {
+    debug new int(1);
+    debug {
+        mayCallGC();
+        auto b = [1, 2, 3];
+        b ~= 4;
+    }
+}
```

Comment:
This can be merged with other compilable tests. Merging them makes the autotester run faster.